docs/tutorials/simple-date-clock-application.adoc

```diff
@@ -0,0 +1,435 @@
+harness-application clockdate -t webapp -t server
+----
+
+The name of the application is `clockdate` and we use the `webapp` and `server` template.
+There is various existing templates with different purpose: for DB interaction, backend, frontend, ...
+
+We observe now that a new directory had been created in the `applications` folder named `clockdate`.
+The folder is organized with many sub-folders, all playing a different role in the app.
+Currently, we will not look at them, we will only run/deploy and access the application at least once.
+
+To do so, we need to generate a specific `helm` configuration (`helm` chart).
+As in the previous section, we use `harness-deployment` for that.
+
+.Generating the `helm` chart for our `clockdate` app
+[source,bash]
+----
+harness-deployment  . -u -dtls -l -d azathoth.local -e local -n azathoth -i clockdate
+----
+
+[NOTE]
+Please note that here we consider that the namespace is already existing. If it doesn't, create it as seen in the previous section.
+
+Now you can build/deploy/run it using `skaffold`.
+
+.Building/deploying/running the webapp with skaffold
+[source,bash]
+----
+skaffold run
+----
+
+Now, you can go to http://clockdate.azathoth.local/ to check your app running!
+In the same time, you can check what the API is answering for the `ping` endpoint on this URL: http://clockdate.azathoth.local/api/ping.
+
+
+=== Modifying your webapp, adding behavior
+
+We are currently capable of generating/running applications, but we did not add our own behavior.
+We need to modify the generated sources to do so.
+If we take a deeper look to the folder generated by `harness-application`, we observe three folders that are the one we will modify on a normal usage/base:
+
+.Generated directory organization
+[source]
+----
++- api               -> owns the OpenAPI definition of the endpoints/resources handled by the API
++- backend
+  `- clockdate       -> the project backend files
+    |- controllers   -> the controller definition
+    `- models        -> the resources exposed by the API
++- fontend           -> the webpage files
+----
+
+In a first time, we will modify the backend to add a new endpoint that will answer in a string the current date and time.
+The process is the following:
+
+. we add the new endpoint in the `openapi` folder, modifying the `openapi.yaml` file,
+. we regenerate the code of the application using `harness-generate`
+. we code the behavior of the endpoint in the dedicated method generated in the `backend/clockdate/controllers` folder.
+. we build/deploy/run the code to see it running (this step can be changed with a pure python run of the backend for a quicker dev loop).
+
+==== Adding the new endpoint to the openapi specification
+
+We will add a new endpoint named `currentdate` that will answer a string when `GET`.
+To do so, we add a special path in the `path` section.
+
+.Modifiying the `api/openapi.yaml` file
+[source,yaml]
+----
```

Comment:
not relevant here, but for you to know something I usually do is to edit openapi specifications on a dedicated tool. I usually go for apicurio

Reply:
Thanks for the pointer! I was wondering if there was some specific tools for that. Matteo also pointed this one. I'll check, otherwise I had in mind to develop a small DSL (perhaps graphic?) to deal with resources definitions. It's perhaps overkill...

Reply:
Apicurio is actually graphical and has all features, and it's open source. What is really missing now is a process to edit/save a file, don't know how that could be achieved. Now I just export/copy the spec manually